frontend/src/components/cluster-dashboard/cluster-graph-view/ClusterGraphView.tsx

```diff
@@ -0,0 +1,233 @@
+import React, { useState, useEffect } from "react";
```

Comment:
General suggestion - if a component hits 200 lines long, that's a rule of thumb where I reconsider if the component might be doing too much, or can be broken apart. This might be an exception, given that working with graphs and charts can have a lot of props to manage. Just something to consider.

Reply:
Gotcha, will keep that in mind.